### ISSUE TITLE
Ansible Galaxy Package Installation Fails with Ansible Base 2.10 due to installing Ansible core (from Ansible-lint requirements)

### ISSUE THREAD

**Issue by jvanderaa** [ansible/ansible-lint#960]
##### Summary

With Ansible 2.10 Base out (`pip install ansible-base`), Ansible Lint will now install Ansible full on top of the Ansible-base, causing an error condition when building a container.  

This is causing Ansible Galaxy package installations to fail.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

This install issue happens with no pinned versions of Ansible or Ansible Lint, getting the latest of Ansible-base and Ansible-Lint.
```console (paste below)
ansible --version
2.10

ansible-lint --version
latest
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

python:3.7.7-buster, python:3.8.5-buster, python:3.7.7-slim-buster

##### STEPS TO REPRODUCE

1. Build a fresh container
2. Install Ansible Base - `pip install ansible-base`
3. Install Ansible-Lint - `pip install ansible-lint`
4. List pip packages installed in the container and there is both ansible-base (2.10), and ansible (core - 2.9.12)

##### Desired Behaviour

I'd expect that ansible-lint would be installed along side only ansible-base, there should not be an install of Ansible core as well.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behaviour

List pip packages installed in the container and there is both ansible-base (2.10), and ansible (core - 2.9.12)

**Comment by ssbarnea** [comment]
I am fully aware of this issue that was caused by Ansible decision to rename the package instead of creating a new one named core.

We cannot fix this without breaking almost all users as they are likely missing to mention ansible as a separated dependency.

I can create two extras "ansible" and "ansible-base" and remove ansible from being listed as a direct dependency, even if we all know it is mandatory. Users will have to update their dependencies to install one of them, the one they prefer.

Still, if we release it with this it would break at least all pre-commit hook users. which are installing only default dependencies.

I am bit inclined to do nothing about that.

**Comment by jvanderaa** [comment]
Sounds good. I will say that in my personal and organizational practices, we look to install Ansible as it's own dependency and add Ansible-Lint on top of it as a separate call out. Following in line with the Python view point of explicit is better than implicit. That would help out. I realize that I don't have the error on here at the moment. Would that be helpful as well?

**Comment by ssbarnea** [comment]
Closing as no action will be done until 2.10 is released, moment where this problem is less likely to be seen.

**Comment by jvanderaa** [comment]
@ssbarnea I'm not sure how this is going to solve the problem of people that use `ansible-base` as their base and want to use ansible-lint. As I understand, Ansible 2.10 has two releases, base which is slimmed down and core which will be what traditional Ansible has been. 

As the product matures and in the near future all of my projects will be installing `ansbile-base`. I do not foresee using Core whatsoever once the projects have moved to 2.10.

Would a PR to remove Ansible from the requirements be accepted? If you are installing Ansible Lint you should already plan on having Ansible installed in some fashion.

**Comment by ssbarnea** [comment]
Yes it is possible but it would require a major version bumping and it will break 9/10 users, and almost all of those that are using ansible-lint as a pre-commit.com hook.

Also all users trying to just install the linter with `pip install ansible-lint` will get a non working linter, as they will likely miss ansible.

**Comment by Thulium-Drake** [comment]
Workaround for now (to make it work with ansible-base only):

```
pip install ansible-lint
pip uninstall ansible -y
pip install ansible-base --force-reinstall
```

Afterwards I was able to run ansible-lint again :-)
